ecommerce/api.py

```diff
@@ -379,3 +379,40 @@ def is_coupon_redeemable(coupon, user):
         return any((mmtrack.has_verified_cert(run.edx_course_key) for run in course.courserun_set.all()))
 
     return True
+
+
+def pick_coupons(user):
+    """
+    Choose the coupon which would be used in redemptions by the user.
+
+    The heuristic is currently:
+     - choose attached coupons over automatic coupons
+     - choose the coupon which has been most recently attached, or most recently modified
+
+    Args:
+        user (django.contrib.auth.models.User): A user
+
+    Returns:
+        Coupon: The coupon which will be used by the user when redeeming runs in a program
+    """
+    sorted_attached_coupons = list(
+        Coupon.objects.filter(usercoupon__user=user).order_by('-usercoupon__updated_on')
+    )
+    sorted_automatic_coupons = list(
+        Coupon.objects.filter(Coupon.is_automatic_qset()).order_by('-updated_on')
+    )
```

Comment:
do you actually need to make a list out of these 2 query sets? I am not sure if it has an actual impact on the performances, but it looks to me that it is unnecessary.